src/main/java/seedu/intern/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,119 @@
+package seedu.intern.logic.parser;
+
+import static seedu.intern.logic.commands.FilterCommand.MESSAGE_NOT_FILTERED;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_COURSE;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_GRADE;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_GRADUATIONYEARMONTH;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_INSTITUTION;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_SKILL;
+import static seedu.intern.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.intern.logic.commands.FilterCommand;
+import seedu.intern.logic.commands.FilterCommand.FilterDescriptor;
+import seedu.intern.logic.parser.exceptions.ParseException;
+import seedu.intern.model.applicant.ApplicationStatus;
+import seedu.intern.model.applicant.Institution;
+import seedu.intern.model.skills.Skill;
+
+/**
+ * Parses input arguments and creates a new FilterCommand object
+ */
+public class FilterCommandParser implements Parser<FilterCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterCommand
+     * and returns a FilterCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterCommand parse(String args) throws ParseException {
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args,
+                        PREFIX_SKILL,
+                        PREFIX_GRADE,
+                        PREFIX_INSTITUTION,
+                        PREFIX_COURSE,
+                        PREFIX_GRADUATIONYEARMONTH,
+                        PREFIX_STATUS);
+
+        FilterDescriptor filterDescriptor = new FilterDescriptor();
+
+        if (argMultimap.getValue(PREFIX_GRADE).isPresent()) {
+            filterDescriptor.setGrade(ParserUtil.parseGrade(argMultimap.getValue(PREFIX_GRADE).get()));
+        }
+        parseInstitutionsForEdit(argMultimap.getAllValues(PREFIX_INSTITUTION))
+                .ifPresent(filterDescriptor::setInstitutions);
+        if (argMultimap.getValue(PREFIX_GRADUATIONYEARMONTH).isPresent()) {
+            filterDescriptor.setGraduationYearMonth(ParserUtil.parseGraduationYearMonth(
+                    argMultimap.getValue(PREFIX_GRADUATIONYEARMONTH).get()));
+        }
+        parseCoursesForEdit(argMultimap.getAllValues(PREFIX_COURSE)).ifPresent(filterDescriptor::setCourses);
+        parseApplicationStatusesForEdit(argMultimap.getAllValues(PREFIX_STATUS))
+                .ifPresent(filterDescriptor::setApplicationStatuses);
+        parseSkillsForEdit(argMultimap.getAllValues(PREFIX_SKILL)).ifPresent(filterDescriptor::setSkills);
+
+        if (!filterDescriptor.isAnyFieldFiltered()) {
```

Comment:
I dont think its an exception since if nothing provided should be equivalent to list all?